### PR TITLE
Make icons squared, and fix build

### LINKED ISSUE #42: Square logo?
I was surprised that the logo is very close to square but it is not square. Is this intentional? Would a logo revision that makes it square be welcome?

### PR DESCRIPTION
**Pull request by samueldr**
Fixes #42.

Closes #118

* * *

Alright, so the whole gist of it is that since even before #42 was opened I already had this problem in mind that needed to be solved.

Some early "don't want to break things" paralysis prevented me from handling this. And I never had since, for many other reasons, including lack of time and energy.

The solution *is* to fix the source SVG files here.

The solution *also* includes dropping everything from the SVG files that was not needed. Those are part of the actual main logo file. Those are artifacts derived from it. So it's fine to drop those "guide" stuff, and even the flat version of the logo.

Also renamed the pair of "snowflake-only" source svg so they are listed alphabetically together. (Also "white" really meant nothing...)

* * *

So, why all those unrelated changes?

To fix the build. I had to make those to test the Makefile changes (from the rename).

The changes don't change the semantics of the existing Makefile, but makes a bare `nix-build` invocation build. Additionally, `nix-shell` now uses facts from `default.nix` instead of duplicating the information and invariably diverging.

### CODE REVIEW

File: bootloader/Makefile [hunk 1]
```diff
@@ -15,3 +16,6 @@ grub2-installer/icons/%.png: grub2-installer/icons/%.svg
 grub2-installer/dejavu.pf2:
 	grub-mkfont -s 20 "$(dejavu)/share/fonts/truetype/DejaVuSans.ttf" -o $@
 
+install: grub2-installer-theme.tar.gz
+	mkdir -p "$(install_dest)"
+	tar --strip-components=1 --directory "$(install_dest)" -xf "$<"
```
Comment (samueldr) [inline bootloader/Makefile:21]: This is silly, but eh, this tests the tar.gz build via the Nix output. If/when CI/CD is done in here, this ensures the tarball is "tested".

**Review by OPNA2608 — COMMENTED** [review]
Resulting icons look fine, and `nix-build` works again. Thanks!

Some non-critical nits, otherwise LGTM.

**Comment by samueldr** [comment]
Thank you for the additional nits!

**Comment by samueldr** [comment]
(I'll note that Nixpkgs has not been changed.)

**Comment by OPNA2608** [comment]
https://github.com/NixOS/nixpkgs/pull/303068 for updating it in Nixpkgs.

**Pull request merged**